warp/src/js_exports/constellation.rs

```diff
@@ -13,3 +16,382 @@ impl ConstellationBox {
+            inner: constellation::file::File::new(name),
+        }
+    }
+    pub fn name(&self) -> String {
+        self.inner.name()
+    }
+    pub fn set_id(&self, id: String) {
+        self.inner.set_id(Uuid::parse_str(&id).unwrap())
+    }
+    pub fn set_name(&self, name: &str) {
+        self.inner.set_name(name)
+    }
+    pub fn description(&self) -> String {
+        self.inner.description()
+    }
+    pub fn set_description(&self, desc: &str) {
+        self.inner.set_description(desc)
+    }
+    pub fn set_thumbnail_format(&self, format: JsValue) {
+        self.inner
+            .set_thumbnail_format(serde_wasm_bindgen::from_value(format).unwrap())
+    }
+    pub fn thumbnail_format(&self) -> JsValue {
+        serde_wasm_bindgen::to_value(&self.inner.thumbnail_format()).unwrap()
+    }
+    pub fn set_thumbnail(&self, data: &[u8]) {
+        self.inner.set_thumbnail(data)
+    }
+    pub fn thumbnail(&self) -> Vec<u8> {
+        self.inner.thumbnail()
+    }
+    pub fn set_favorite(&self, fav: bool) {
+        self.inner.set_favorite(fav)
+    }
+    pub fn favorite(&self) -> bool {
+        self.inner.favorite()
+    }
+    pub fn set_reference(&self, reference: &str) {
+        self.inner.set_reference(reference)
+    }
+    pub fn set_thumbnail_reference(&self, reference: &str) {
+        self.inner.set_thumbnail_reference(reference)
+    }
+    pub fn reference(&self) -> Option<String> {
+        self.inner.reference()
+    }
+    pub fn thumbnail_reference(&self) -> Option<String> {
+        self.inner.thumbnail_reference()
+    }
+    pub fn size(&self) -> usize {
+        self.inner.size()
+    }
+    pub fn set_size(&self, size: usize) {
+        self.inner.set_size(size)
+    }
+    pub fn set_creation(&self, creation: js_sys::Date) {
+        self.inner.set_creation(creation.into())
+    }
+    pub fn set_modified(&self, modified: Option<js_sys::Date>) {
+        self.inner.set_modified(modified.map(|d| d.into()))
+    }
+    pub fn hash(&self) -> Hash {
+        self.inner.hash()
+    }
+    pub fn set_hash(&self, hash: Hash) {
+        self.inner.set_hash(hash)
+    }
+    pub fn set_file_type(&self, file_type: JsValue) {
+        self.inner
+            .set_file_type(serde_wasm_bindgen::from_value(file_type).unwrap())
+    }
+    pub fn file_type(&self) -> JsValue {
+        serde_wasm_bindgen::to_value(&self.inner.file_type()).unwrap()
+    }
+    pub fn path(&self) -> String {
+        self.inner.path().to_string()
+    }
+    pub fn set_path(&mut self, new_path: &str) {
+        self.inner.set_path(new_path)
+    }
+    pub fn id(&self) -> String {
+        self.inner.id().to_string()
+    }
+    pub fn creation(&self) -> js_sys::Date {
+        self.inner.creation().into()
+    }
+    pub fn modified(&self) -> js_sys::Date {
+        self.inner.modified().into()
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Clone)]
+pub struct Item {
+    file: Option<File>,
+    directory: Option<Directory>,
+}
```

Comment:
ditto

Reply:
If we are to use this route, couldnt it have been best to use `Item` in `constellation` instead?

Reply:
Yes, thanks! I've made the change.